lib/SILGen/SILGenFunction.cpp

```diff
@@ -638,6 +638,54 @@ void SILGenFunction::emitGeneratorFunction(SILDeclRef function, Expr *value) {
   emitEpilog(Loc);
 }
 
+void SILGenFunction::emitGeneratorFunction(SILDeclRef function, VarDecl *var) {
+  MagicFunctionName = SILGenModule::getMagicFunctionName(function);
+
+  RegularLocation loc(var);
+  loc.markAutoGenerated();
+
+  auto decl = function.getAbstractFunctionDecl();
+  auto *dc = decl->getInnermostDeclContext();
+  auto interfaceType = var->getInterfaceType();
```

Comment:
Can you test weak or unowned properties? I suspect you might need to change this to getValueInterfaceType() to strip off the reference storage qualifier.